webpack.dev.config.js

```diff
@@ -0,0 +1,17 @@
+const path = require('path');
+const { createConfig } = require('@edx/frontend-build');
+/** Uncomment the lines below if you wish to use an env.config.js in development */
+// const envConfig = require('./env.config');
```

Comment:
[curious, suggestion] If you uncomment this without an `env.config.js` file defined in the repository, does it fail? Assuming not due to frontend-build's `moduleNameMapper` configuration. I believe if it's not defined, `envConfig` would fallback as an empty object `{}`.

If that is true, in the commented out line below, e.g., could you have it such that its not necessary to have commented out code here? For example, if you do want to use `env.config.js` as is and I uncomment this line, I wouldn't feel I can commit it upstream since not everyone might have an `env.config.js`, which adds some burden to local development to ensure I don't commit these changes.

Would `config.devServer.port = envConfig?.PORT || process.env.PORT || 8080;` work while keeping the import uncommented?

Reply:
It'll fail on this line because it'll first try to find this file before it gets to line 6 (`createConfig('webpack-dev')`. Unless I add some condition before to check that the file exists, then I imagine it would need to remain uncommented like this (but I totally agree on the local dev burden to not commit these changes) or we escalate the frontend-build ticket to handle this PORT assignment for us. Thoughts?

Reply:
Ah, yes I see what you mean...

I do imagine we could get a ✅ fairly quickly if we opened a PR in frontend-build upstream and request a review in FWG, as the ideal path forward, IMHO.

That said, an alternative shorter term idea might be to (temporarily) utilize `patch-package` ([docs](https://www.npmjs.com/package/patch-package)) to make a patch to your installed copy of `@edx/frontend-build` that is committed and applied anytime `npm install` is run (including in CI/GoCD).

FWIW, we also had to utilize `patch-package` in an enterprise MFE semi-recently at one point, given we were blocked on upgrading to latest version of Paragon but needed a fix in one of its components for the currently installed version we were stuck on (see [this ADR](https://github.com/openedx/frontend-app-admin-portal/blob/d457357b184f7e72330e8745b165abec8dc0f4e7/docs/decisions/0007-patch-package.rst) describing the rationale).

Reply:
Based on your suggestions, I moved the webpack logic for both prod and dev to a patch file, and I also started working on a branch in `frontend-build`[ to include these upgrades](https://github.com/openedx/frontend-build/tree/jwesson/add-js-config-handling-to-webpack-jest). 
You'll also notice in the patch file that I modified the webpack and jest configs to be able to handle a JS or JSX extension of `env.config`, which I know we'll want to be able to have for upcoming work.